core/src/workflows/author_entry.rs

```diff
@@ -31,9 +31,14 @@ pub async fn author_entry<'a>(
     // 2. Validate the entry
     await!(validate_entry(entry.clone(), validation_data, &context))?;
     // 3. Commit the entry
-    await!(commit_entry(entry.clone(), maybe_crud_link, &context))?;
+    let addr = await!(commit_entry(entry.clone(), maybe_crud_link, &context))?;
     // 4. Publish the valid entry to DHT. This will call Hold to itself
-    await!(publish(entry.address(), &context))
+    //TODO: missing a general public/private sharing check here, for now just manually
+    // forcing CapTokenGrant to private
+    if entry.entry_type() != EntryType::CapTokenGrant {
```

Comment:
There is already a function `EntryType::can_publish()` where currently it just distinguishes `EntryType::Dna` from everything else (see https://github.com/holochain/holochain-rust/blob/develop/core_types/src/entry/entry_type.rs#L91). At least this check should go there. When I said that this is is broken I meant for App entries where we actually have to check the DNA to see how it is defined. I had left some code commented out there that was used in the DHT reducer before adding the network.

Reply:
fixed